### PR TITLE
feat: Add freeze all assets spell, integration for all assets (SC-243)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: test/IntegrationTests.t.sol [hunk 1]
```diff
@@ -41,141 +44,421 @@ contract IntegrationTests is Test {
...
+                untestedReserves.push(asset);
+                continue;
+            }
 
-        assertTrue(authority.hat() == spell);
-    }
+            // Max out supply caps for this asset so that they aren't hit during test
+            vm.startPrank(SPARK_PROXY);
+            poolConfig.setSupplyCap(asset, 68_719_476_735);  // MAX_SUPPLY_CAP
+            poolConfig.setBorrowCap(asset, 68_719_476_735);  // MAX_BORROW_CAP
```
Comment (hexonaut) [inline test/IntegrationTests.t.sol:407]: These can just be set to `0` to disable.

File: test/IntegrationTests.t.sol [hunk 2]
```diff
@@ -41,141 +44,421 @@ contract IntegrationTests is Test {
...
 
-        vm.prank(randomUser);  // Demonstrate no ACL in spell
-        freezeWethSpell.freeze();
+            // Setup spell, max out supply caps so that they aren't hit during test
+            _vote(freezeAssetSpell);
 
-        assertEq(_isFrozen(WETH), true);
+            vm.startPrank(SPARK_PROXY);
+            poolConfig.setSupplyCap(asset, 68_719_476_735);  // MAX_SUPPLY_CAP
+            poolConfig.setBorrowCap(asset, 68_719_476_735);  // MAX_BORROW_CAP
```
Comment (hexonaut) [inline test/IntegrationTests.t.sol:324]: These can just be set to `0` to disable.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Some changes, but also just realized the spell should have a flag set so that it can only be fired once. We will deploy a new spell for each time we want to use this. Otherwise users can freeze the market over and over until the hat changes. Additional tests should be added to ensure the precrafted spells can only execute at most once.

Also, the spell unfreezing wbtc and dai ltv to 0 will execute tomorrow, so let's get those changes added before merging this as well.

File: test/IntegrationTests.t.sol (hunk 1 above)
Comment (lucas-manuel) [reply test/IntegrationTests.t.sol]: Updated

File: test/IntegrationTests.t.sol (hunk 2 above)
Comment (lucas-manuel) [reply test/IntegrationTests.t.sol]: Updated

**Review by hexonaut — CHANGES_REQUESTED** [review]
Small change then good to merge.

File: src/spells/EmergencySpell_SparkLend_FreezeAllAssets.sol [hunk 3]
```diff
@@ -7,12 +7,16 @@ contract EmergencySpell_SparkLend_FreezeAllAssets {
...
+    bool public executed;
+
     constructor(address sparklendFreezerMom_) {
         sparkLendFreezerMom = sparklendFreezerMom_;
     }
 
     function freeze() external {
+        require(!executed, "FreezeAllAssetsSpell/already-executed");
         ISparkLendFreezerMom(sparkLendFreezerMom).freezeAllMarkets();
+        executed = true;
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/spells/EmergencySpell_SparkLend_FreezeAllAssets.sol:19]: Updated

**Review by hexonaut — APPROVED** [review]
Good to merge.

**Pull request merged**